examples/seismic/acoustic/operators.py

```diff
@@ -15,13 +14,21 @@ def freesurface(model, eq):
     eq : Eq
         Time-stepping stencil (time update) to mirror at the freesurface.
     """
-    lhs, rhs = eq.evaluate.args
+    lhs, rhs = eq.args
     # Get vertical dimension and corresponding subdimension
-    zfs = model.grid.subdomains['fsdomain'].dimensions[-1]
+    fsdomain = model.grid.subdomains['fsdomain']
+    zfs = fsdomain.dimensions[-1]
     z = zfs.parent
 
-    # Functions present in the stencil
-    funcs = retrieve_functions(rhs)
+    # Retrieve vertical derivatives
```

Comment:
what's all these new additions?

Reply:
it does the same, except it only evaluates `z` derivatives instead of everything since the rest isn't needed which is much cheaper especially for things like tti.